protocols/imap/src/main/java/org/apache/james/imap/api/process/ImapSession.java

```diff
@@ -233,4 +233,10 @@ default Username getUserName() {
             .map(MailboxSession::getUser)
             .orElse(null);
     }
+
+    default boolean isPlainAuthDisabledEventually() {
```

Comment:
Can we try to improve this method name?

Also does this method still make sense when we implement OIDC using IMAP SASL?

```
Given plainAuth disabled
 -> With OIDC and no SSL AUTH should fail
 -> With OIDC and SSL auth should succeed
 -> With PLAIN and no SSL AUTH should fail
 -> With PLAIN and SSL auth should fail
 ```
 
 The condition is not the case in both case.
 
 Sorry for proposing to factorize code, in the end, given short term evolutions it likely do not make much sense...

Reply:
Can you tell me more about this? I am not sure that I understand what you mean here. On my side, this method covers only for the PLAIN case and not for the ODIC case. For the ODIC case I think we need a new condition anyway.